intelmq/bots/outputs/cif3/output.py

```
@@ -0,0 +1,233 @@
+
+        matched_tag = False
+        if 'classification.type' in intelmq_event:
+            for classification in INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP.keys():
+                if classification in intelmq_event['classification.type']:
+                    new_cif3_dict['tags'].extend(
+                        INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP[classification].split(','))
+                    matched_tag = True
+            if not matched_tag:
+                new_cif3_dict['tags'].append(intelmq_event['classification.type'])
+
+        for new_tag in self.cif3_additional_tags:
+             new_cif3_dict['tags'].append(new_tag)
+
+        # map the confidence
+        if 'feed.accuracy' in intelmq_event:
+            if not self.cif3_static_confidence:
+                new_cif3_dict['confidence'] = (intelmq_event['feed.accuracy'] / 10)
+        if not new_cif3_dict.get('confidence'):
+            new_cif3_dict['confidence'] = self.cif3_feed_confidence
+
+
+        # map remaining IntelMQ fields to CIFv3 fields
+        for intelmq_type in INTELMQ_TO_CIF_FIELDS_MAP.keys():
+            if intelmq_type in intelmq_event:
+                cif3_field = INTELMQ_TO_CIF_FIELDS_MAP[intelmq_type]
+                new_cif3_dict[cif3_field] = intelmq_event[intelmq_type]
+
+        # build the CIFv3 indicator object from the dict
+        new_indicator = None
+        try:
+            new_indicator = Indicator(**new_cif3_dict)
+        except Exception as err:
+            self.logger.error(f"Error creating indicator: {err}")
+        
+        return new_indicator
+        
+
+
+    def _submit_cif3_indicator(self, indicators):
+        # build the CIFv3 indicator object from the dict
+        try:
+            self.logger.info(f"Sending {len(indicators)} indicator(s).")
+            resp = self.cli.indicators_create(indicators)
+        except Exception as err:
+            self.logger.error(f"Error submitting indicator(s): {err}")
```

That just logs and ignores the error, the data is removed from IntelMQ (acknowledged).
Is that your intention? If yes, why? One of IntelMQ's principles is to never loose any data (even if there is an error message).

---

Would it be appropriate to re-raise `err` in the exception after logging the error?

---

Yes, that works

---

i've made those changes changes in commit 500f6d5ede3a48cef101c0952c17306eeed72412